tests/rules/avoid-duplicate-actions-in-reducer.test.ts

```diff
@@ -1,176 +1,183 @@
+import type { InvalidTestCase } from '@typescript-eslint/experimental-utils/dist/ts-eslint'
```

Comment:
```suggestion
import type { TSESLint } from '@typescript-eslint/experimental-utils'
```

We definetly should use the [`@typescript-eslint/no-restricted-imports`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-restricted-imports.md) (available since [`@typescript-eslint@4.32.0`](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v4.32.0)) to make sure we don't commit this mistake again as it's a common one (references I've been involved in 😄: https://github.com/timdeschryver/eslint-plugin-ngrx/pull/158, https://github.com/angular-eslint/angular-eslint/pull/260#discussion_r547471168, https://github.com/typescript-eslint/typescript-eslint/issues/2931 and https://github.com/angular-eslint/angular-eslint/pull/711).

Reply:
I just realized that `@typescript-eslint/no-restricted-imports` won't help us right now as there's a bug there that it was not reporting `type-only` imports properly (I sent a small fix there though, PR: https://github.com/typescript-eslint/typescript-eslint/pull/3996).

For now, we could use [**`no-restricted-imports`**](https://eslint.org/docs/rules/no-restricted-imports) instead and change it when updating to `@typescript-eslint@v5`.